crates/proof-of-sql/src/base/commitment/test_commitment.rs

```diff
@@ -0,0 +1,113 @@
+use std::fmt::Debug;
+use std::ops::{AddAssign, Neg, Sub, SubAssign};
+
+use curve25519_dalek::RistrettoPoint;
+use serde::{Deserialize, Serialize};
+
+use crate::base::commitment::CommittableColumn;
+use crate::base::scalar::test_scalar::TestScalar;
+
+use super::Commitment;
+
+/// This should only be used for the purpose of unit testing.
+/// For now this is simply decorating RistrettoPoint so as to make a struct that
+/// functions the same but is viewed as a different struct to the compiler.
+#[derive(Copy, Clone)]
```

Comment:
That does work. I don't know think that would have necessarily been the same thing. I was trying to use RistrettoPoint's implementations of those traits, rather than whatever the default implementations are.